servers/visual/visual_server_scene.cpp

```diff
@@ -2294,20 +2294,18 @@ _FORCE_INLINE_ static void _light_capture_sample_octree(const RasterizerStorage:
 				half >>= 1;
 			}
 
-			if (cell == RasterizerStorage::LightmapCaptureOctree::CHILD_EMPTY) {
-				alpha[c][n] = 0;
-			} else {
+			if (cell != RasterizerStorage::LightmapCaptureOctree::CHILD_EMPTY) {
 				alpha[c][n] = p_octree[cell].alpha;
 
 				for (int i = 0; i < 6; i++) {
 					//anisotropic read light
 					float amount = p_dir.dot(aniso_normal[i]);
-					if (amount < 0) {
-						amount = 0;
+
+					if (amount > 0) {
+						color[c][n].x += p_octree[cell].light[i][0] * 0.0009765f * amount;
+						color[c][n].y += p_octree[cell].light[i][1] * 0.0009765f * amount;
+						color[c][n].z += p_octree[cell].light[i][2] * 0.0009765f * amount;
```

Comment:
Huh, does this have real gain? I would assume something like `/1024` is directly optimized to be `*1/1024` -> `*0.0009765` somewhere while Godot compiles, and this seems far more confusing.

Reply:
I don't know why but I tested it again and patch with ```/ 1024.0``` is 48% faster and with ```* 0.0009765f``` is 57% faster than non-patched code!

Reply:
It'd be nice to keep the perf gains without scarifying readability. Seeing the final value it's difficult to deduce the same meaning as /1024.

maybe a constant value could be used?

Reply:
You can use `constexpr float ONE_1024TH = 1.0 / 1024.0;`.

Reply:
> You can use `constexpr float ONE_1024TH = 1.0 / 1024.0;`.

I used this and it's same as ```* 0.0009765f```, thanks.
I pushed changes.